### DISCUSSION TITLE
infiniteloading 组件

### DISCUSSION THREAD

**Discussion by yangxiaolu1993** [jdf2e/nutui-react#11] [General]
1、组件内定义的 className 值未使用  BEM 规范

2、代码73行： 使用 useEffect 挂载 scroll 事件时，为什么设置 hasMore, isInfiniting 更新时触发？

`useEffect(() => {
    const parentElement = getParentElement(scroller.current as HTMLDivElement) as Node & ParentNode
    scrollEl.current = useWindow ? window : parentElement
    scrollEl.current.addEventListener('scroll', handleScroll, useCapture)
    return () => {
      scrollEl.current.removeEventListener('scroll', handleScroll, useCapture)
    }
  }, [hasMore, isInfiniting])`

3、 需要支持 className 与 style

**Comment by junjun666** [comment]
@yangxiaolu1993 
1.已修复
2.因为在滚动监听的handleScroll函数中，有两个变量hasMore和isInfiniting依赖，如果不放在useEffect的依赖中，这两个变量不会实时更新的(react hooks机制）。
3.已新增支持 className 与 style

**Comment by bruceleedrupal** [comment]
为何直接测试的demo都无法正常载入组件，而且一直没有解决

**Comment by xiaoyatong** [comment]
1. 是否考虑防抖和节流。
2. 关于 useWindow 和 useCapture的命名，这个想法很有意思，但是建议规避hooks的这套方式，除非它确实是实现了hooks~
